### ISSUE TITLE
missing result type with cast to slice child

### ISSUE THREAD

**Issue by nektro** [ziglang/zig#16683]
### Zig Version

0.11.0

### Steps to Reproduce and Observed Behavior

```zig
const std = @import("std");

test {
    var x: f32 = 1.0;
    try std.testing.expectFmt("ff", "{}", .{std.fmt.fmtSliceHexLower(&.{@intFromFloat(x * 255.0)})});
}
```

```
test.zig:5:73: error: @intFromFloat must have a known result type
    try std.testing.expectFmt("ff", "{}", .{std.fmt.fmtSliceHexLower(&.{@intFromFloat(x * 255.0)})});
                                                                        ^~~~~~~~~~~~~~~~~~~~~~~~
test.zig:5:73: note: use @as to provide explicit result type
```

### Expected Behavior

all test passed, result type should be taken from parameter which is set as `[]const u8`

**Comment by mlugg** [comment]
Would be solved by (and afaict essentially can *only* be solved by) #16512

**Comment by xdBronch** [comment]
this is fixed on master
